### PR TITLE
refactor: upgrade to Gorm 2.0

### PR DESCRIPTION
**Pull request by b0undl3ss**
This PR updates the Gorm and Gormigrate packages to v2. The postgres driver used by Gorm v1 is in maintenance mode and had issues with dropping connections and not being able to recover. 

I ran the integration tests and they passed without error, and I haven't run into any issues in my environment.

### CODE REVIEW

File: pkg/bastion/dbinit.go [hunk 1]
```diff
@@ -661,15 +660,15 @@ func DBInit(db *gorm.DB) error {
 		ErrMsg: "sshportal was halted while the connection was still active",
 	}).Error
 }
-
+/* Implemented hard delete via Unscoped() method in Gorm library
```
Comment (moul) [inline pkg/bastion/dbinit.go:663]: Can you completely delete the old implementation?

File: go.mod [hunk 2]
```diff
@@ -30,6 +28,10 @@ require (
 	golang.org/x/term v0.0.0-20210422114643-f5beecf764ed // indirect
 	golang.org/x/tools v0.1.7
 	gopkg.in/gormigrate.v1 v1.6.0
```
Comment (moul) [inline go.mod:30]: can you check why we still have this dep here?

**Comment by moul** [comment]
Thank you very much for this PR,

Please be sure to run the various auto-linter tools like `go mod tidy`, `gofmt -w`, etc; it should help to fix the last CI red flags

Thanks! 🙏

**Comment by b0undl3ss** [comment]
@moul Thank you for the feedback, fixing things up now!

File: pkg/bastion/dbinit.go (hunk 1 above)
Comment (b0undl3ss) [reply pkg/bastion/dbinit.go]: Removed in 6acdc3c

File: go.mod (hunk 2 above)
Comment (b0undl3ss) [reply go.mod]: Resolved in 9cee4e8

**Comment by b0undl3ss** [comment]
Investigating unit test failures...

**Comment by b0undl3ss** [comment]
Issues have all been resolved. Let me know if there's anything else I need to do to get this merged. Thank you!

**Review by moul — APPROVED** [review]

**Pull request merged**